### PR TITLE
add tensor support for gaussian_random_op test=develop

### PR DESCRIPTION
**Pull request by wangchaochaohu**
As https://github.com/PaddlePaddle/Paddle/pull/23056
相比https://github.com/PaddlePaddle/Paddle/pull/23056  未删除gaussian_random_batch_size_like，单纯支持Tensor Shape
image

image

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -10000,33 +10000,55 @@ def gaussian_random(shape, mean=0.0, std=1.0, seed=0, dtype='float32'):
     Generate a random tensor whose data is drawn from a Gaussian distribution.
 
     Args:
-        shape (Tuple[int] | List[int]): Shape of the generated random tensor.
-
+        shape (Tuple[int] | List[int] | Tensor | TensorList): Shape of the generated random tensor.
```
Comment (liym27) [inline python/paddle/fluid/layers/nn.py]: 1. 首字母小写
Tuple -> tuple
List -> list

2. Tensor不是python 数据类型
Tensor -> Variable
TensorList -> list[Variable]

File: paddle/fluid/operators/gaussian_random_op.cc [hunk 2]
```diff
@@ -94,7 +138,18 @@ class GaussianRandomOpMaker : public framework::OpProtoAndCheckerMaker {
 
     AddAttr<std::vector<int64_t>>("shape",
                                   "(vector<int64_t>) "
-                                  "The dimension of random tensor.");
+                                  "The dimension of random tensor.")
+        .SetDefault({});
```
Comment (liym27) [inline paddle/fluid/operators/gaussian_random_op.cc:151]: LGTM for setting default value here.

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (wangchaochaohu) [reply python/paddle/fluid/layers/nn.py]: Done

File: paddle/fluid/operators/fill_constant_op.h [hunk 3]
```diff
@@ -26,7 +26,8 @@ namespace operators {
 
 using Tensor = framework::Tensor;
 
-inline framework::DDim GetShape(const framework::ExecutionContext &ctx) {
+inline framework::DDim GetShape(const framework::ExecutionContext &ctx,
+                                std::string op_type) {
```
Comment (liym27) [inline paddle/fluid/operators/fill_constant_op.h:31]: A suggestion： use `GetDataFromTensor` and `GetDataFromTensorList` to compatible with `int32` and `int64`

File: paddle/fluid/operators/gaussian_random_op.cc [hunk 4]
```diff
@@ -58,12 +86,18 @@ class GaussianRandomOp : public framework::OperatorWithKernel {
     for (auto dim : shape) {
       temp.push_back(static_cast<int64_t>(dim));
     }
-    PADDLE_ENFORCE_GT(
-        shape.size(), 0UL,
-        platform::errors::InvalidArgument(
-            "Attribute(shape) of GaussianRandomOp must be set "
-            "and shape.size() > 0, but reveived shape.size() is %d",
-            shape.size()));
```
Comment (liym27) [inline paddle/fluid/operators/gaussian_random_op.cc:66]: Better not to delete this `PADDLE_ENFORCE_GT `

File: paddle/fluid/operators/fill_constant_op.h (hunk 3 above)
Comment (wangchaochaohu) [reply paddle/fluid/operators/fill_constant_op.h:31]: Done

File: paddle/fluid/operators/gaussian_random_op.cc (hunk 4 above)
Comment (wangchaochaohu) [reply paddle/fluid/operators/gaussian_random_op.cc:66]: Done

**Review by liym27 — APPROVED** [review]
LGTM

**Review by Heeenrrry — APPROVED** [review]
LGTM

**Pull request merged**